### PR TITLE
fix(api-server): service-insights should never return items: null

### LINKED ISSUE #6639: Some collection resource endpoints return items as null instead of an empty array
### What happened?

There seem to be at least some collection resource endpoints for which an empty list of resources is returned as `items: null` instead of `items: []`.

I could only find meshes/default/service-insights, but for this endpoint at least, it’s true.

All collection resource endpoints in Kuma should behave the same and return an empty list as `[]`.

### PR DESCRIPTION
**Pull request by lahabana**
Our api should never return null instead of an empty array for lists. This fixes an issue in service-insights

Fix #6639

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/api-server/service_insight_endpoints.go [hunk 1]
```diff
@@ -90,24 +94,27 @@ func (s *serviceInsightEndpoints) listResources(request *restful.Request, respon
 // 2) Mesh+Name is a key on Universal, but not on Kubernetes, so if there are two services of the same name in different Meshes we would have problems with naming.
 // From the API perspective it's better to provide ServiceInsight per Service, not per Mesh.
 // For this reason, this method expand the one ServiceInsight resource for the mesh to resource per service
-func (s *serviceInsightEndpoints) expandInsights(serviceInsightList *mesh.ServiceInsightResourceList) rest.ResourceList {
-	restItems := []*rest_unversioned.Resource{}
+func (s *serviceInsightEndpoints) expandInsights(serviceInsightList *mesh.ServiceInsightResourceList) []rest.Resource {
+	restItems := []rest.Resource{} // Needs
```
Comment (jakubdyszkiewicz) [inline pkg/api-server/service_insight_endpoints.go:98]: Needs?

Comment (lahabana) [reply pkg/api-server/service_insight_endpoints.go:98]: isn't it obvious :)

File: pkg/api-server/service_insight_endpoints.go [hunk 2]
```diff
@@ -71,8 +72,11 @@ func (s *serviceInsightEndpoints) listResources(request *restful.Request, respon
 		return
 	}
 
-	restList := s.expandInsights(serviceInsightList)
-	restList.Total = uint32(len(restList.Items))
+	items := s.expandInsights(serviceInsightList)
+	restList := rest.ResourceList{
+		Total: uint32(len(items)),
+		Items: items,
```
Comment (michaelbeaumont) [inline pkg/api-server/service_insight_endpoints.go:78]: Shouldn't we just set `json:"omitempty"` on all ResourceLists?

**Review by jakubdyszkiewicz — APPROVED** [review]

File: pkg/api-server/service_insight_endpoints.go (hunk 2 above)
Comment (lahabana) [reply pkg/api-server/service_insight_endpoints.go:78]: I don't think so it's a little weird to have this field omitted from a list response and get: `{"total": 0, "next": null}`

Comment (michaelbeaumont) [reply pkg/api-server/service_insight_endpoints.go:78]: Right if we set `items []SomeResource \`json:"Items,omitempty"\`` we wouldn't ever get `items: null`, would we? Or maybe I'm missing something

Comment (michaelbeaumont) [reply pkg/api-server/service_insight_endpoints.go:78]: Ah you're saying it should never be omitted. Ok, i guess that's also a valid API guarantee. Arg golang

**Pull request merged**